docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,157 @@
+    },
+    "universe": {
+        "initFile": "A simple file created with the `init` subcommand"
+    },
+    "error": false,
+    "errorMsgs": []
+}
+```
+
+In this example, the `init` command of the plugin has created a new file called `initFile`. 
+
+The content of this file is: `A simple file created with the init subcommand`, which is recorded in the `universe` field of the response.
+
+This output is then sent back to Kubebuilder, allowing it to incorporate the changes made by the plugin into the project.
+
+<aside class="note">
+<h1>Caution</h1>
+
+When writing your own external plugin, you **should not** directly echo or print anything to the stdout. 
+
+This is because Kubebuilder and your plugin are communicating with each other via `stdin` and `stdout` using structured `JSON` data.
+Any additional information sent to stdout (such as debug messages or logs) that's not part of the expected PluginResponse JSON structure may cause parsing errors when Kubebuilder tries to read and decode the response from your plugin.
+
+If you need to include logs or debug messages while developing your plugin, consider writing these messages to a log file instead.
+
+</aside>
+
+## How to use it?
+
+### Prerequisites
+- kubebuilder CLI > 3.11.0
+- An executable for the external plugin.
+
+  This could be a plugin that you've created yourself, or one from an external source.
+- Configuration of the external plugin's path.
+
+  This can be done by setting the `${EXTERNAL_PLUGINS_PATH}` environment variable, or by placing the plugin in a path that follows a `group-like name and version` scheme:
+```sh
+# for Linux
+$HOME/.config/kubebuilder/plugins/${name}/${version}
+
+# for OSX
+~/Library/Application Support/kubebuilder/plugins/${name}/${version}
+```
+
+### Subcommands:
+
+The external plugin supports the same subcommands as kubebuilder already provides:
+- `init`: project initialization.
+- `create api`: scaffold Kubernetes API definitions.
+- `create webhook`: scaffold Kubernetes webhooks.
+- `edit`: update the project configuration.
+
+Also, there are **Optional** subcommands for a better user experience:
+- `metadata`: add customized plugin description and examples when a `--help` flag is specified. 
+- `flags`: specify valid flags for Kubebuilder to pass to the external plugin.
+
+<aside class="note">
+<h1>More about `flags` subcommand</h1>
+
+The `flags` subcommand in an external plugin allows for early error detection by informing Kubebuilder about the flags the plugin supports. If an unsupported flag is identified, Kubebuilder can issue an error before the plugin is called to execute. 
+If a plugin does not implement the `flags` subcommand, Kubebuilder will pass all flags to the plugin, making it the external plugin's responsibility to handle any invalid flags. 
+
+If a plugin does implement the `flags` subcommand, it **MUST** support `--init`, `--api`, `--webhook`, and `--edit` flags, which correspond to respective Kubebuilder subcommands.
```

Comment:
Nit: Now that I am thinking about it a bit more I don't think these flags are _required_ to be implemented as the external plugin would return an error about an unknown flag that Kubebuilder would ignore. That being said, we may want to consider the pros and cons of enforcing or not enforcing this behavior. 

@Eileen-Yu I'll leave it up to you if you want to change the wording to reflect that these flags aren't _required_ . IMO though this nit shouldn't block this PR from merging.

Reply:
I can understand that flag for `--init` … is more like a ‘bonus’ instead of ‘must’. For me, actually I’m still curious about the detailed scenario to support that so as to reach “fast fail”.

So I guess maybe we can drop that paragraph includes the 'must' literal.
I was imagining if it is fair enough to stay focus on the general usage(`init`, `create` and `edit`) of the external plugin.